REFDOC.md

```diff
@@ -0,0 +1,156 @@
+Return a tuple `(snap_posts_scanned, snap_scan_time, snap_posts_per_second)`.
+- `reset_stat()`: Obtain `rw_lock`. Set `num_posts_scanned` and `post_scan_time` to `0`. Release `rw_lock`.
+- `reset_snap()`: Obtain `rw_lock`. Set `snap_num_posts_scanned` and `snap_post_scan_time` to `0`. Release `rw_lock`.
+#### Considerations  
+- Both read and write operations are lock protected,
+as it is necessary to perform thread safe write and to prevent reading while another thread is writing.
+### Known bugs  
+- None
+> Last change in this section was on 19 May 2020.  
+## Class GlobalVars.MSStatus  
+Tracking metasmoke status.  
+### Public interface  
+- `is_up()`: Query if metasmoke is up. Returns `True` if metasmoke is up and `False` otherwise.
+- `is_down()`: Query if metasmoke is down. Returns `True` if metasmoke is down and `False` otherwise.
+- `failed()`: Indicate a metasmoke connection failure.
+- `succeeded()`: Indicate a metasmoke connection success.
+- `get_failure_count()`: Get metasmoke connection failure counter value. The counter counts consecutive failures.
+- `reset_ms_status()`: Reset class `GlobalVars.ms_status` to default values.
+### Thread safety  
+Yes.  
+### Notes on usage  
+- Metasmoke counter represents consecutive metasmoke failures.
+### Implementation  
+This class is implemented with 2 static variables and 1 lock.  
+#### Details  
+##### Attributes  
+- `ms_is_up`: Whether or not metasmoke is up.
+- `counter`: Metasmoke connection failure counter.
+- `rw_lock`: Lock. Controlling access to `ms_is_up`, `failure_count` and `last_ping_time`.
+##### Methods  
+- `set_up()`: Private to `metasmoke.py`. Obtain `rw_lock`. Set `ms_is_up` to `True`. Release `rw_lock`.
+- `set_down()`: Private to `metasmoke.py`. Obtain `rw_lock`. Set `ms_is_up` to `False`. Release `rw_lock`.
+- `is_up()`: Obtain `rw_lock`. Read `ms_is_up` into `current_ms_status`. Release `rw_lock`. Return `current_ms_status`.
+- `is_down()`: Call `is_up()` and return the inverted returned value.
+- `failed()`: Obtain `rw_lock`. Increase `counter` by `1`. Release `rw_lock`.
+- `succeeded()`: Obtain `rw_lock`. Set `counter` to `0`. Release `rw_lock`.
+- `get_failure_count()`: Obtain `rw_lock`. Read `counter` into `current_counter`. Release `rw_lock`. Return `current_failure_count`.
+- `reset_ms_status`: Obtain `rw_lock`. Set `ms_is_up` to `True`. Set `counter` to `0`. Release `rw_lock`.
+#### Considerations  
+- `is_down()` is implemented to call `is_up()`, so maintainance tasks only need to be performed on `is_up()`.
+### Known bugs  
+- None.
+> Last change in this section was on 20 May 2020.
+# File metasmoke.py  
+## Class Metasmoke  
+> Note: the documentation of this class is not yet completed.
+### Subclasses  
+- `AutoSwitch`
+### Public interface  
+- `ms_up()`: Switch metasmoke status to up.
```

Comment:
These two should probably be `set_ms_up()` and `set_ms_down()`. With them not having `set_` (or something similar), it's ambiguous as to if they change the state or return the current state.